frog/imports/client/SingleActivity/index.js

```diff
@@ -9,20 +10,38 @@ import ConfigPanel from './ConfigPanel';
 import Loading from './Loading';
 import Finish from './Finish';
 import { type StateT } from './types';
+import { activityTypes } from '/imports/activityTypes';
+import { templatesObj } from '/imports/internalTemplates';
 
 /**
  * The main class for the Single Activity
  */
 class SingleActivity extends React.Component<{}, StateT> {
   constructor() {
     super();
+
     this.state = {
       stage: 1
     };
   }
 
+  componentDidMount = () => {
+    InjectData.getData('duplicate', data => {
+      if (data) {
+        const type =
+          activityTypes.find(x => x.id === data.activityType) ||
+          templatesObj[data.activityType];
+        this.setState({
+          stage: 2,
+          activityType: type,
+          config: data.config
+        });
+      }
+    });
+  };
```

Comment:
We should isolate this call outside of `index.js` to limit the number of responsibilities SingleActivity has.